docker/publish.sh

```diff
@@ -4,7 +4,43 @@ set -euo pipefail
 # Cd into directory holding this script
 cd "${BASH_SOURCE[0]%/*}"
 
-image=${1:-"flexflow-cuda"}
+# Default values
+cuda_version=$(nvcc --version | grep "release" | awk '{print $NF}')
+# Change cuda_version eg. V11.7.99 to 11.7
+cuda_version=${cuda_version:1:4}
+image="flexflow-cuda"
+
+# Parse command-line options
+while [[ $# -gt 0 ]]; do
+  key="$1"
+
+  case $key in
+    --cuda_version)
+      cuda_version="$2"
+      shift 2
+      ;;
+    --image_name)
+      image="$2"
+      shift 2
+      ;;
+    *)
+      echo "Invalid option: $key"
+      exit 1
+      ;;
+  esac
+done
+
+if [[ "$cuda_version" != @(11.1|11.3|11.5|11.6|11.7|11.8) ]]; then
+  # validate the verison of CUDA against a list of supported ones
+  # 11.1, 11.3, 11.5, 11.6, 11.7, 11.8
+  echo "cuda_version is not supported, please choose among {11.1,11.3,11.5,11.6,11.7,11.8}"
+  exit 1
+fi
+
+# modify cuda version to xx.x.0
+cuda_version=${cuda_version}.0
```

Comment:
same problem as above with regards to versions of cuda for which nvidia does not publish the xx.x.0 docker image